src/app/components/SelectWithSearch.tsx

```diff
@@ -13,9 +14,12 @@ type SelectWithSearchProps = {
 }
 
 export default function SelectWithSearch(props: SelectWithSearchProps) {
+
     const [search, setSearch] = useState("")
     const [selected, setSelected] = useState(props.selected)
 
+    const { handleSetCountry } = useStorage();
```

Comment:
SelectWithSearch is a generic component that should not know anything about what it's selecting. Please abstract away the concept of country to keep the component as generic as possible.